src/units/abilities.cpp

```diff
@@ -2095,6 +2095,105 @@ bool attack_type::special_matches_filter(const config & cfg, const std::string&
 	return common_matches_filter(cfg, tag_name, filter);
 }
 
+bool attack_type::has_special_with_filter(const config & filter) const
+{
+	using namespace utils::config_filters;
+	bool simple_check = !filter["active"].to_bool();
+	for(const auto [key, cfg] : specials().all_children_view()) {
+		if(special_matches_filter(cfg, key, filter)){
+			if(simple_check){
+				return true;
+			}
+			if ( special_active(cfg, AFFECT_SELF, key) ) {
+				return true;
+			}
+		}
+	}
+	// Skip checking the opponent's attack?
```

Comment:
Why is this comment a question? Is there uncertainty about what's being done?

Reply:
comment removed, it's was a copy/past of has_special() comment